pandas/core/arrays/string_.py

```diff
@@ -79,37 +84,114 @@ class StringDtype(ExtensionDtype):
     StringDtype
     """
 
-    name = "string"
-
     #: StringDtype.na_value uses pandas.NA
     na_value = libmissing.NA
+    _metadata = ("storage",)
+
+    def __init__(self, storage=None):
+        if storage is None:
+            storage = get_option("mode.string_storage")
+        if storage not in {"python", "pyarrow"}:
+            raise ValueError(
+                f"Storage must be 'python' or 'pyarrow'. Got {storage} instead."
+            )
+        self.storage = storage
```

Comment:
We probably want to raise an ImportError when trying to create this dtype with `storage="arrow"` if pyarrow is not installed?

Reply:
is this comment addresses?

Reply:
no, we check if pyarrow installed in the array construction

```
>>> pd.StringDtype(storage="pyarrow")
string[pyarrow]
>>> 
>>> pd.Series(list("abc"), dtype="string[pyarrow]")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/simon/pandas/pandas/core/series.py", line 442, in __init__
    data = sanitize_array(data, index, dtype, copy)
  File "/home/simon/pandas/pandas/core/construction.py", line 552, in sanitize_array
    subarr = _try_cast(data, dtype, copy, raise_cast_failure)
  File "/home/simon/pandas/pandas/core/construction.py", line 728, in _try_cast
    subarr = array_type(arr, dtype=dtype, copy=copy)
  File "/home/simon/pandas/pandas/core/arrays/string_arrow.py", line 155, in _from_sequence
    cls._chk_pyarrow_available()
  File "/home/simon/pandas/pandas/core/arrays/string_arrow.py", line 149, in _chk_pyarrow_available
    raise ImportError(msg)
ImportError: pyarrow>=1.0.0 is required for PyArrow backed StringArray.
>>> 
>>> pd.Series(list("abc"), dtype=pd.StringDtype(storage="pyarrow"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/simon/pandas/pandas/core/series.py", line 442, in __init__
    data = sanitize_array(data, index, dtype, copy)
  File "/home/simon/pandas/pandas/core/construction.py", line 552, in sanitize_array
    subarr = _try_cast(data, dtype, copy, raise_cast_failure)
  File "/home/simon/pandas/pandas/core/construction.py", line 728, in _try_cast
    subarr = array_type(arr, dtype=dtype, copy=copy)
  File "/home/simon/pandas/pandas/core/arrays/string_arrow.py", line 155, in _from_sequence
    cls._chk_pyarrow_available()
  File "/home/simon/pandas/pandas/core/arrays/string_arrow.py", line 149, in _chk_pyarrow_available
    raise ImportError(msg)
ImportError: pyarrow>=1.0.0 is required for PyArrow backed StringArray.
>>> 
```

Reply:
This PR parameterizes the dtype, so changing this behavior here is out of scope. opened #41732 with added tests